kerastuner/engine/oracle.py

```diff
@@ -17,90 +17,241 @@
+          `trial_id`: The id for this Trial.
 
         Returns:
-            A dictionary with keys:
-                - status: string, one of "RUN", "IDLE", "EXIT"
-                - values: Only included if status is "RUN".
-                    Dict mapping parameter names to suggested values.
-                    Note that if the Oracle is keeping tracking of a large
-                    space, it may return values for more parameters
-                    than what was listed in `space`.
+            A dictionary with keys "values" and "status", where "values" is
+            a mapping of parameter names to suggested values, and "status"
+            is the TrialStatus that should be returned for this trial (one
+            of "RUNNING", "IDLE", or "STOPPED").
         """
         raise NotImplementedError
 
-    def result(self, trial_id, score):
-        """Record the measured objective for a set of parameter values.
+    def create_trial(self, tuner_id):
+        # Allow for multi-worker DistributionStrategy within a Trial.
+        if tuner_id in self.ongoing_trials:
+            return self.ongoing_trials[tuner_id]
 
-        If not overridden, this method does nothing.
+        trial_id = trial_lib.generate_trial_id()
 
-        Args:
-            trial_id: String. Unique id for this trial.
-            score: Scalar. Lower is better.
-        """
-        pass
+        if len(self.trials.items()) >= self.max_trials:
+            status = trial_lib.TrialStatus.STOPPED
+            values = None
+        else:
+            response = self.populate_space(trial_id)
+            status = response['status']
+            values = response['values'] if 'values' in response else None
+
+        hyperparameters = self.hyperparameters.copy()
+        hyperparameters.values = values
+        trial = trial_lib.Trial(
+            hyperparameters=hyperparameters,
+            trial_id=trial_id,
+            status=status)
 
-    def report_status(self, trial_id, status, score=None, t=None):
-        """Used by a worker to report the current status of a trial.
+        if status == trial_lib.TrialStatus.RUNNING:
+            self.ongoing_trials[tuner_id] = trial
+            self.trials[trial_id] = trial
+
+        return trial
+
+    def update_trial(self, trial_id, metrics, t=0):
+        """Used by a worker to report the status of a trial.
 
         Args:
             trial_id: A previously seen trial id.
-            status: String, one of "RUNNING", "CANCELLED". A status of
-                "CANCELLED" means a trial has crashed or been deemed
-                infeasible.
-            score: (Optional) Float. The current, intermediate value of the
-                trial's objective.
-            t: (Optional) Float. The current value in a timeseries representing
-                the state of the trial. This is the value that `score` will be
-                associated with.
+            metrics: Dict of float. The current value of this
```

Comment:
Tensorflow 1.0 / Tensorboard support string metrics. Is this not the case for Keras?

Reply:
Keras only has float metrics. What would we need string metrics for?

Reply:
SG, yeah Keras only has numeric metrics, TF summaries are handled a little differently